### PR TITLE
feat(edit-page): Edit content shortcut when edit url content map page

### LINKED ISSUE #27561: Edit Page V2: Provide Shortcut to Edit Content
### Parent
#27546 

### User Story
As a content editor, I want to be able to easily edit a piece of content that generates a page, so I can make updates efficiently.

https://github.com/dotCMS/core/assets/751424/e960a772-bead-4618-9993-1bb95bac100a

```[tasklist]
### Acceptance Criteria
- [ ] A button is added to the page editor interface. (UI coming soon)
- [ ] Clicking the button opens a dialog to edit the corresponding piece of content.
- [ ] After making changes in the dialog, clicking "Save" closes the dialog and refreshes the page.
```

### Proposed Objective
Static content: "Core Features"

### Proposed Priority
Static content: "Priority 3 - Average"

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
Static content: "N/A"

### Assumptions & Initiation Needs
- The location of the actual content is unknown.
- The button can be added dynamically to the page editor interface.

### Quality Assurance Notes & Workarounds
No additional notes at this time.

### PR DESCRIPTION
**Pull request by rjvelazco**
### Video

https://github.com/dotCMS/core/assets/72418962/c4d0284b-71c2-447a-866d-d725bca1c4cb

### CODE REVIEW

**Review by zJaaal — APPROVED** [review]

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts [hunk 1]
```diff
@@ -577,31 +589,41 @@ export class EditEmaEditorComponent implements OnInit, OnDestroy {
...
+
+                if (!didInsert) {
+                    this.handleDuplicatedContentlet();
+
+                    return;
                 }
+
+                // Save when created
+                this.store.savePage({
+                    pageContainers,
```
Comment (manuelrojas) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts:619]: what happen if the savePage fails? does it shows an error?

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.html [hunk 2]
```diff
@@ -90,7 +97,7 @@
             }">
             <iframe
                 #iframe
-                [src]="(es.clientHost ? es.iframeURL : '') | safeUrl"
+                [src]="es.iframeURL | safeUrl"
```
Comment (KevinDavilaDotCMS) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.html:100]: I think `"(es.clientHost ? es.iframeURL : '') | safeUrl"` is needed to load vtl pages, why the change here?

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts (hunk 1 above)
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts:619]: The store method has a hander. In that case, we show an empty state! @manuelrojas

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.html (hunk 2 above)
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.html:100]: While editing a VTL page, the `iframeURL` is empty. Therefore, I removed the conditional.

Is there any case when is a VTL Page and we have a `iframeURL`? @KevinDavilaDotCMS

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts [hunk 3]
```diff
@@ -577,31 +589,41 @@ export class EditEmaEditorComponent implements OnInit, OnDestroy {
...
-                            this.reloadIframe();
-                        }
-                    });
-                } else {
+                if (!payload) {
                     this.reloadIframe(); // We still need to reload the iframe because the contentlet is not in the container yet
+
+                    return;
+                }
+
```
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts:605]: So let me see if I get this:

First we reload the store

Then we reload the iframe to get the new data.

I know store and iframe are independent a little bit, but can this be a race condition?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts (hunk 3 above)
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts:605]: @fmontes each `if` statement has its `return`. There shouldn't be a race condition

File: core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-ema-dialog/store/dot-ema-dialog.store.ts [hunk 4]
```diff
@@ -34,6 +34,12 @@ export interface CreateFromPaletteAction {
     payload: ActionPayload;
 }
 
+interface EditContentI {
```
[the remark this answers is not in the document]
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-ema-dialog/store/dot-ema-dialog.store.ts:37]: Done!

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts (hunk 3 above)
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts]: Done!

File: core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-ema-dialog/store/dot-ema-dialog.store.ts [hunk 5]
```diff
@@ -122,13 +128,15 @@ export class DotEmaDialogStore extends ComponentStore<EditEmaDialogState> {
      *
      * @memberof DotEmaDialogStore
      */
-    readonly editContentlet = this.updater((state, payload: { inode: string; title: string }) => {
+    readonly editContentlet = this.updater((state, { inode, title, isUrlMap }: EditContentI) => {
```
[the remark this answers is not in the document]
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-ema-dialog/store/dot-ema-dialog.store.ts]: Done!

**Review by fmontes — APPROVED** [review]

**Review by KevinDavilaDotCMS — APPROVED** [review]
